app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/drawer/DrawerViewModel.java

```diff
@@ -0,0 +1,262 @@
+import com.teamagam.gimelgimel.domain.rasters.SetIntermediateRasterInteractorFactory;
+import com.teamagam.gimelgimel.domain.user.repository.UserPreferencesRepository;
+import devlight.io.library.ntb.NavigationTabBar;
+import java.util.ArrayList;
+import java.util.List;
+
+@AutoFactory
+public class DrawerViewModel extends BaseViewModel<MainActivityDrawer> {
+
+  private final DisplayVectorLayersInteractorFactory mDisplayVectorLayersInteractorFactory;
+  private final SetVectorLayerVisibilityInteractorFactory
+      mSetVectorLayerVisibilityInteractorFactory;
+  private final DisplayIntermediateRastersInteractorFactory
+      mDisplayIntermediateRastersInteractorFactory;
+  private final SetIntermediateRasterInteractorFactory mSetIntermediateRasterInteractorFactory;
+  private final DisplayUserLocationsInteractorFactory mDisplayUserLocationsInteractorFactory;
+
+  private final UserPreferencesRepository mUserPreferencesRepository;
+  private final Context mContext;
+
+  private DisplayVectorLayersInteractor mDisplayVectorLayersInteractor;
+  private DisplayIntermediateRastersInteractor mDisplayIntermediateRastersInteractor;
+  private DisplayUserLocationsInteractor mDisplayUserLocationsInteractor;
+
+  private RecyclerViewAdapterSetter mAdapterSetter;
+
+  private UserLocationsRecyclerAdapter mUsersAdapter;
+  private LayersRecyclerAdapter mBubbleLayersAdapter;
+  private LayersRecyclerAdapter mLayersAdapter;
+  private RastersRecyclerAdapter mRastersAdapter;
+
+  public DrawerViewModel(
+      @Provided DisplayVectorLayersInteractorFactory displayVectorLayersInteractorFactory,
+      @Provided SetVectorLayerVisibilityInteractorFactory setVectorLayerVisibilityInteractorFactory,
+      @Provided
+          DisplayIntermediateRastersInteractorFactory displayIntermediateRastersInteractorFactory,
+      @Provided SetIntermediateRasterInteractorFactory setIntermediateRasterInteractorFactory,
+      @Provided DisplayUserLocationsInteractorFactory displayUserLocationsInteractorFactory,
+      @Provided UserPreferencesRepository userPreferencesRepository,
+      Context context,
+      RecyclerViewAdapterSetter adapterSetter) {
+    mDisplayVectorLayersInteractorFactory = displayVectorLayersInteractorFactory;
+    mSetVectorLayerVisibilityInteractorFactory = setVectorLayerVisibilityInteractorFactory;
+    mDisplayIntermediateRastersInteractorFactory = displayIntermediateRastersInteractorFactory;
+    mSetIntermediateRasterInteractorFactory = setIntermediateRasterInteractorFactory;
+    mDisplayUserLocationsInteractorFactory = displayUserLocationsInteractorFactory;
+    mUserPreferencesRepository = userPreferencesRepository;
+    mContext = context;
+    mAdapterSetter = adapterSetter;
+  }
+
+  @Override
+  public void start() {
+    super.start();
+    initializeDisplayInteractors();
+    initializeAdapters();
+  }
+
+  public String getUsername() {
+    return mUserPreferencesRepository.getString(mContext.getString(R.string.user_name_text_key));
```

Comment:
We're using it instead of the SharedPreferences just so we won't rewrite the code.